Remove some uses of WTF_MAKE_TZONE_ALLOCATED_INLINE.

MenloDorian opened the pull request:
#### 6bb793baab557706eacafa4358b88ee56e9e0672
<pre>
Remove some uses of WTF_MAKE_TZONE_ALLOCATED_INLINE.
<a href="https://bugs.webkit.org/show_bug.cgi?id=283922">https://bugs.webkit.org/show_bug.cgi?id=283922</a>
<a href="https://rdar.apple.com/140800944">rdar://140800944</a>

Reviewed by Simon Fraser.

They shouldn&apos;t have been used in header files.  This fix is needed to speed up builds.

Also made the following clean up changes:
1. Removed WTF_MAKE_COMPACT_TZONE_ALLOCATED_INLINE and WTF_MAKE_STRUCT_TZONE_ALLOCATED_INLINE
   which are no longer used.
2. Make sure that all config.h files #include &lt;wtf/TZoneMalloc.h&gt;.

* Source/WTF/config.h:
* Source/WTF/wtf/TZoneMallocInlines.h:
* Source/WebCore/Modules/airplay/WebMediaSessionManager.cpp:
* Source/WebCore/Modules/cache/DOMCacheStorage.cpp:
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDM.cpp:
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDMPrivate.h:
* Source/WebCore/Modules/fetch/FetchLoader.cpp:
* Source/WebCore/Modules/fetch/FetchLoaderClient.h:
* Source/WebCore/Modules/fetch/FetchResponse.cpp:
* Source/WebCore/Modules/fetch/FetchResponseBodyLoader.h:
* Source/WebCore/Modules/identity/CredentialRequestCoordinator.cpp:
* Source/WebCore/Modules/identity/CredentialRequestCoordinatorClient.h:
* Source/WebCore/Modules/indexeddb/server/IDBBackingStore.h:
* Source/WebCore/Modules/indexeddb/server/IDBServer.cpp:
* Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabaseManager.h:
* Source/WebCore/Modules/mediastream/DetachedRTCDataChannel.h:
* Source/WebCore/Modules/mediastream/RTCDataChannel.cpp:
* Source/WebCore/Modules/reporting/ReportingObserver.cpp:
* Source/WebCore/Modules/reporting/ReportingObserverCallback.h:
* Source/WebCore/Modules/storage/DummyStorageProvider.h:
* Source/WebCore/Modules/webaudio/AudioWorkletProcessor.cpp:
* Source/WebCore/Modules/webaudio/AudioWorkletProcessorConstructionData.h:
* Source/WebCore/Modules/webauthn/AuthenticatorCoordinator.cpp:
* Source/WebCore/Modules/webauthn/AuthenticatorCoordinatorClient.h:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/config.h:
* Source/WebCore/PAL/pal/CMakeLists.txt:
* Source/WebCore/PAL/pal/HysteresisActivity.h:
* Source/WebCore/PAL/pal/PALTZoneImpls.cpp: Copied from Source/WebCore/dom/TransformSourceLibxslt.cpp.
* Source/WebCore/PAL/pal/crypto/CryptoDigest.h:
* Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.cpp:
* Source/WebCore/PAL/pal/system/Clock.h:
* Source/WebCore/PAL/pal/system/ClockGeneric.cpp:
* Source/WebCore/PAL/pal/system/ClockGeneric.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXObjectCache.cpp:
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AXTextMarker.cpp:
* Source/WebCore/accessibility/AXTextMarker.h:
* Source/WebCore/accessibility/AXTreeStore.h:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.cpp:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.h:
* Source/WebCore/bindings/js/BindingsJSTZoneImpls.cpp: Copied from Source/WebCore/dom/TransformSourceLibxslt.cpp.
* Source/WebCore/bindings/js/CachedScriptSourceProvider.h:
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:
* Source/WebCore/bindings/js/ScriptBufferSourceProvider.h:
* Source/WebCore/bindings/js/WebAssemblyCachedScriptSourceProvider.h:
* Source/WebCore/bindings/js/WebAssemblyScriptBufferSourceProvider.h:
* Source/WebCore/config.h:
* Source/WebCore/crypto/CryptoAlgorithmParameters.h:
* Source/WebCore/crypto/CryptoAlgorithmTZoneImpls.cpp: Added.
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCbcCfbParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCtrParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesGcmParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcdhKeyDeriveParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcdsaParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmHkdfParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmHmacKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmPbkdf2Params.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaHashedImportParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaKeyGenParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaOaepParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaPssParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmX25519Params.h:
* Source/WebCore/css/CSSGroupingRule.cpp:
* Source/WebCore/css/CSSKeyframesRule.cpp:
* Source/WebCore/css/CSSRuleList.h:
* Source/WebCore/css/CSSStyleRule.cpp:
* Source/WebCore/css/calc/CSSCalcTree.cpp:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/dom/ActiveDOMObject.cpp:
* Source/WebCore/dom/DataTransfer.cpp:
* Source/WebCore/dom/DeviceMotionClient.h:
* Source/WebCore/dom/DeviceMotionController.cpp:
* Source/WebCore/dom/DeviceOrientationClient.h:
* Source/WebCore/dom/DeviceOrientationController.cpp:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/EventLoop.cpp:
* Source/WebCore/dom/Node.cpp:
* Source/WebCore/dom/PopoverData.cpp:
* Source/WebCore/dom/PopoverData.h:
* Source/WebCore/dom/RadioButtonGroups.cpp:
* Source/WebCore/dom/ScriptExecutionContext.cpp:
* Source/WebCore/dom/ScriptExecutionContext.h:
* Source/WebCore/dom/TransformSource.h:
* Source/WebCore/dom/TransformSourceLibxslt.cpp:
* Source/WebCore/dom/WindowOrWorkerGlobalScopeTrustedTypes.cpp:
* Source/WebCore/editing/EditingStyle.cpp:
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
* Source/WebCore/fileapi/Blob.cpp:
* Source/WebCore/fileapi/BlobLoader.h:
* Source/WebCore/loader/EmptyClients.cpp:
* Source/WebKit/config.h:
* Source/bmalloc/bmalloc/TZoneHeap.h:
* Tools/DumpRenderTree/config.h:
* Tools/TestWebKitAPI/config.h:

Canonical link: <a href="https://commits.webkit.org/287322@main">https://commits.webkit.org/287322@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/87c975d994ae17db55e11c6a79eadd5330bcc104

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/79125 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58160 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/32501 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83759 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30331 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/81258 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/67263 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/6427 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/61928 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19832 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/82192 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/51966 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/72111 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42233 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/49317 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/26229 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28702 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/72246 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/70433 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/26651 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/85150 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/78337 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/6442 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/4492 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/70159 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6605 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/67982 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69407 "Found 4 new API test failures: /TestWebKit:WebKit.PageLoadDidChangeLocationWithinPage, /TestWebKit:WebKit.EnumerateDevicesCrash, /TestWebKit:WebKit.PageLoadBasic, /TestWebKit:WebKit2UserMessageRoundTripTest.WKURL (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13456 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/12258 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/100662 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12223 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/6394 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/12204 "Built successfully") | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/21966 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/struct-new_default-small-members.js.wasm-bbq (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/6346 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/9804 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8138 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WTF/wtf/PlatformUse.h:
```diff
@@ -307,7 +307,7 @@
 #if !defined(USE_TZONE_MALLOC)
 #if (CPU(ARM64) || CPU(X86_64)) && OS(DARWIN) && (__SIZEOF_POINTER__ == 8)
-#define USE_TZONE_MALLOC 0 /* temporarily disable FIXME rdar://140325747 */
+#define USE_TZONE_MALLOC 1 /* temporarily disable FIXME rdar://140325747 */
```
Is this intentional? If so, you can remove this comment.

MenloDorian replied:
This is intentional only for testing.  Still trying to figure out how to make this PR a Draft.  It's not ready for review

smfr commented on Source/WTF/config.h:
```diff
@@ -29,6 +29,7 @@
 #undef new
 #undef delete
 #include <wtf/FastMalloc.h>
+#include <wtf/TZoneMalloc.h>
```
It would be nice to have data to tell us if this actually helps.

MenloDorian replied:
This one is not currently needed.  I included it for consistency so that anyone adding uses of TZoneHeap macros  in WTF code won't be surprised why their code doesn't build.  If you have data showing that this severely impacts build speed, I can omit it for the WTF case for now.

Commit:
Remove some uses of WTF_MAKE_TZONE_ALLOCATED_INLINE.
https://bugs.webkit.org/show_bug.cgi?id=283922
rdar://140800944

Reviewed by Simon Fraser.

They shouldn't have been used in header files.  This fix is needed to speed up builds.

Also made the following clean up changes:
1. Removed WTF_MAKE_COMPACT_TZONE_ALLOCATED_INLINE and WTF_MAKE_STRUCT_TZONE_ALLOCATED_INLINE
   which are no longer used.
2. Make sure that all config.h files #include <wtf/TZoneMalloc.h>.

* Source/WTF/config.h:
* Source/WTF/wtf/TZoneMallocInlines.h:
* Source/WebCore/Modules/airplay/WebMediaSessionManager.cpp:
* Source/WebCore/Modules/cache/DOMCacheStorage.cpp:
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDM.cpp:
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDMPrivate.h:
* Source/WebCore/Modules/fetch/FetchLoader.cpp:
* Source/WebCore/Modules/fetch/FetchLoaderClient.h:
* Source/WebCore/Modules/fetch/FetchResponse.cpp:
* Source/WebCore/Modules/fetch/FetchResponseBodyLoader.h:
* Source/WebCore/Modules/identity/CredentialRequestCoordinator.cpp:
* Source/WebCore/Modules/identity/CredentialRequestCoordinatorClient.h:
* Source/WebCore/Modules/indexeddb/server/IDBBackingStore.h:
* Source/WebCore/Modules/indexeddb/server/IDBServer.cpp:
* Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabaseManager.h:
* Source/WebCore/Modules/mediastream/DetachedRTCDataChannel.h:
* Source/WebCore/Modules/mediastream/RTCDataChannel.cpp:
* Source/WebCore/Modules/reporting/ReportingObserver.cpp:
* Source/WebCore/Modules/reporting/ReportingObserverCallback.h:
* Source/WebCore/Modules/storage/DummyStorageProvider.h:
* Source/WebCore/Modules/webaudio/AudioWorkletProcessor.cpp:
* Source/WebCore/Modules/webaudio/AudioWorkletProcessorConstructionData.h:
* Source/WebCore/Modules/webauthn/AuthenticatorCoordinator.cpp:
* Source/WebCore/Modules/webauthn/AuthenticatorCoordinatorClient.h:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/config.h:
* Source/WebCore/PAL/pal/CMakeLists.txt:
* Source/WebCore/PAL/pal/HysteresisActivity.h:
* Source/WebCore/PAL/pal/PALTZoneImpls.cpp: Copied from Source/WebCore/dom/TransformSourceLibxslt.cpp.
* Source/WebCore/PAL/pal/crypto/CryptoDigest.h:
* Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.cpp:
* Source/WebCore/PAL/pal/system/Clock.h:
* Source/WebCore/PAL/pal/system/ClockGeneric.cpp:
* Source/WebCore/PAL/pal/system/ClockGeneric.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXObjectCache.cpp:
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AXTextMarker.cpp:
* Source/WebCore/accessibility/AXTextMarker.h:
* Source/WebCore/accessibility/AXTreeStore.h:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.cpp:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.h:
* Source/WebCore/bindings/js/BindingsJSTZoneImpls.cpp: Copied from Source/WebCore/dom/TransformSourceLibxslt.cpp.
* Source/WebCore/bindings/js/CachedScriptSourceProvider.h:
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.h:
* Source/WebCore/bindings/js/ScriptBufferSourceProvider.h:
* Source/WebCore/bindings/js/WebAssemblyCachedScriptSourceProvider.h:
* Source/WebCore/bindings/js/WebAssemblyScriptBufferSourceProvider.h:
* Source/WebCore/config.h:
* Source/WebCore/crypto/CryptoAlgorithmParameters.h:
* Source/WebCore/crypto/CryptoAlgorithmTZoneImpls.cpp: Added.
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCbcCfbParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCtrParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesGcmParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcdhKeyDeriveParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcdsaParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmHkdfParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmHmacKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmPbkdf2Params.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaHashedImportParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaKeyGenParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaOaepParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaPssParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmX25519Params.h:
* Source/WebCore/css/CSSGroupingRule.cpp:
* Source/WebCore/css/CSSKeyframesRule.cpp:
* Source/WebCore/css/CSSRuleList.h:
* Source/WebCore/css/CSSStyleRule.cpp:
* Source/WebCore/css/calc/CSSCalcTree.cpp:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/dom/ActiveDOMObject.cpp:
* Source/WebCore/dom/DataTransfer.cpp:
* Source/WebCore/dom/DeviceMotionClient.h:
* Source/WebCore/dom/DeviceMotionController.cpp:
* Source/WebCore/dom/DeviceOrientationClient.h:
* Source/WebCore/dom/DeviceOrientationController.cpp:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/EventLoop.cpp:
* Source/WebCore/dom/Node.cpp:
* Source/WebCore/dom/PopoverData.cpp:
* Source/WebCore/dom/PopoverData.h:
* Source/WebCore/dom/RadioButtonGroups.cpp:
* Source/WebCore/dom/ScriptExecutionContext.cpp:
* Source/WebCore/dom/ScriptExecutionContext.h:
* Source/WebCore/dom/TransformSource.h:
* Source/WebCore/dom/TransformSourceLibxslt.cpp:
* Source/WebCore/dom/WindowOrWorkerGlobalScopeTrustedTypes.cpp:
* Source/WebCore/editing/EditingStyle.cpp:
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
* Source/WebCore/fileapi/Blob.cpp:
* Source/WebCore/fileapi/BlobLoader.h:
* Source/WebCore/loader/EmptyClients.cpp:
* Source/WebKit/config.h:
* Source/bmalloc/bmalloc/TZoneHeap.h:
* Tools/DumpRenderTree/config.h:
* Tools/TestWebKitAPI/config.h:

Canonical link: https://commits.webkit.org/287322@main
